### PR TITLE
Feat/kumactl access token

### LINKED ISSUE #2041: Support access token on kumactl
Kuma-cp may be protected behind a oauth server. Kumactl would need support for it to be configured with access token. Access token can be created out of band and supplied as argument to kumactl.

### PR DESCRIPTION
**Pull request by sudeeptoroy**
### Summary

feat(kumactl) added --add-header command line arg

### Full changelog

A generic way to add any header while kumaclt connects to kuma-cp.

### Issues resolved

Fix #2041

### CODE REVIEW

File: app/kumactl/cmd/config/config_control_planes_add.go [hunk 1]
```diff
@@ -76,6 +90,7 @@ func newConfigControlPlanesAddCmd(pctx *kumactl_cmd.RootContext) *cobra.Command
 	cmd.Flags().StringVar(&args.clientKeyFile, "client-key-file", "", "path to the certificate key of a client that is authorized to use the Admin operations of the Control Plane (kumactl stores only a reference to this file)")
 	cmd.Flags().StringVar(&args.caCertFile, "ca-cert-file", "", "path to the certificate authority which will be used to verify the Control Plane certificate (kumactl stores only a reference to this file)")
 	cmd.Flags().BoolVar(&args.skipVerify, "skip-verify", false, "skip CA verification")
+	args.addHeaders = cmd.Flags().StringArray("add-header", nil, "add these headers while communicating to control plane")
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/config/config_control_planes_add.go:93]: There is `StringToStringVar` take a look at `install control plane`, so you can just receive the map and don't have to parse it yourself

Comment (jakubdyszkiewicz) [reply app/kumactl/cmd/config/config_control_planes_add.go:93]: (the format then will be `--add-header abc=value`)

File: pkg/util/http/client.go [hunk 2]
```diff
@@ -16,12 +18,15 @@ func (f ClientFunc) Do(req *nethttp.Request) (*nethttp.Response, error) {
 	return f(req)
 }
 
-func ClientWithBaseURL(delegate Client, baseURL *url.URL) Client {
+func ClientWithBaseURL(delegate Client, baseURL *url.URL, apiServer *config_proto.ControlPlaneCoordinates_ApiServer) Client {
```
Comment (jakubdyszkiewicz) [inline pkg/util/http/client.go:21]: how about a `headers map[string]string` this way we can avoid `pkg/util` to depend on kumactl config

File: pkg/config/app/kumactl/v1alpha1/config.proto [hunk 3]
```diff
@@ -48,6 +52,9 @@ message ControlPlaneCoordinates {
     // ClientKey defines the key of the authorized client of the control plane
     // API server
     string client_key_file = 4;
+
+    // Headers to be added for communication with Kuma control plane
+    repeated AddHeader add_headers = 5;
```
Comment (jakubdyszkiewicz) [inline pkg/config/app/kumactl/v1alpha1/config.proto:57]: maybe just 
```
repeated Headers headers = 5;
```
and the argument `--header key=value`?

**Review by jakubdyszkiewicz — COMMENTED** [review]
some minor stuff but overall 👍

File: pkg/config/app/kumactl/v1alpha1/config.proto (hunk 3 above)
Comment (sudeeptoroy) [reply pkg/config/app/kumactl/v1alpha1/config.proto:57]: ok

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**